infrastructures/infrastructure-azure/src/main/java/org/ow2/proactive/resourcemanager/nodesource/billing/AzureBillingRateCard.java

```diff
@@ -126,45 +126,59 @@ String getRateCard(AzureBillingCredentials azureBillingCredentials) throws IOExc
         return null;
     }
 
-    public void updateVmRates(AzureBillingCredentials azureBillingCredentials)
+    // synchronized to ensure we dont try to use meter rates to compute the resources cost while we are updating them
+    synchronized public HashMap<String, LinkedHashMap<String, Double>> updateOrGetMetersRates(
+            AzureBillingCredentials azureBillingCredentials, HashSet<String> metersIdsSet, boolean update)
```

Comment:
I think split it into two function updateMetersRates and getMetersRates could be easier to understand and maintain. Is there a reason they are put into one function ?

Reply:
You re totally right.. 2 synchronized methods of the same object share the same lock (as you explained to me) and i am in this situation.